src/ui/window.cpp

```diff
@@ -759,6 +765,32 @@ void Window::limitSize(int* w, int* h)
   *h = std::max(*h, border().height());
 }
 
+void Window::limitPosition(int* x, int* y, int* w, int* h)
```

Comment:
Prefer returning a gfx::Point or gfx::Rect, and if w and h can be accessed via `bounds()`, remove these params (anyway just making these comments without further information, I saw that this PR will introduced several things later)

Reply:
My first thought was to follow how the limitSize function is defined, but I will see if I can refactor it.

Reply:
Oh, and this private function is used just in the commit that contains it, there is no plan to use it later.

Reply:
I believe I would just leave this as is, because this function just wants to limit the values of the parameters, which are the future values for the window's position and size. So I cannot use bounds(), and returning a Point or Rect would be inconvenient IMO.

Reply:
Avoid using raw pointers and try to use gfx::Rect or gfx::Point if it's possible. There are still some functions using raw int* pointers in the code, are from the C era and should be refactored, I will highly recommend you to avoid using raw pointers (they are too error prone, needs a lot of `if (nonNull)` checks, and make the code a lot harder to read).

Reply:
Just as an example, you can use something like this:
```c++
          gfx::Point pos = clickedWindowPos->origin() + mousePos - clickedMousePos;

          // Limit window position to avoid hiding the title bar
          limitPosition(pos);
```
And `limitPosition(gfx::Point& pos)`.

Something that I don't like from the original code, is that there is no check for `clickedWindowPos != nullptr` (not even an `assert`). And it came from [the very first import to Git code](https://github.com/aseprite/aseprite/blob/57e978b102f3ae0924d47b5f6b5f7d1305933363/jinete/src/jwindow.c#L346) (C code, year 2007).

Reply:
Oh I didn't want to modify the original code, or the least possible at least...now I'm going to update it, thanks.